### ISSUE TITLE
Publishing a dataset with USDC 6 decimal `publisherMarketFeeToken` have wrong `publishMarketFeeAmount` stored in subgraph

### ISSUE THREAD

**Issue by krisliew** [oceanprotocol/market#1911]
**Describe the bug**
Hi there I am trying to publish an asset with `publisherMarketOrderFee` of  `'1'` using `USDC` as the `publisherMarketFeeToken`. After publishing, I realised that the outcome value for `publishMarketFeeAmount` stored in subgraph is wrong. Hence, the dataset cannot be consumed by anyone.

**To Reproduce**
1. Start a localhost marketplace  
2. Go to `app.config.js`, update `publisherMarketOrderFee = '1'`
3. Go to `src/components/Publish/_utils.ts`, look for `const ercParams` and update the `feeToken: '0x07865c6E87B9F70255377e024ace6630C1Eaa37F'` (6 decimal USDC Address)
4. Start the marketplace in localhost, and publish a fixed price asset
5. Go to the [published dataset page](https://github.com/oceanprotocol/market/issues/new?assignees=&labels=Type%3A+Bug&template=bug_report.md&title=)
6. Open Network tab in developer tool (right-click inspect element), look for subgraph response
7. check the `publishMarketFeeAmount` as in the screenshot

**Expected behavior**
1. `publishMarketFeeAmount` should be `1` for dataset published using `USDC` as the `publisherMarketFeeToken`

**Screenshots**
As in the screenshot below, we can observe that `publisherMarketFeeAmount` is equal to `1000000000000` for the response from subgraph.
![image](https://user-images.githubusercontent.com/39853992/233248595-ec40d9f7-e9c0-4aa2-bd09-ef8389e7ec11.png)
Dataset Link:
- https://market.oceanprotocol.com/asset/did:op:d1831e870809147e7ec2aad2b679cccea541aa4b5577aa43c85c2da0392118a9

`publisherMarketFeeToken` (Goerli USDC address): 
- https://goerli.etherscan.io/token/0x07865c6E87B9F70255377e024ace6630C1Eaa37F 

**Desktop (please complete the following information):**
 - Mac OS
 - Brave v1.49.120

**Additional context**
- I am suspecting this is a decimal value issue somewhere around the subgraph else the contract where the implementation only support `18` decimal ERC20 address.

**Comment by jamiehewitt15** [comment]
Yeah, this is potentially an issue on the subgraph rather than the market.

**Comment by soonhuat** [comment]
@jamiehewitt15 should this move to subgraph repo issue?
